src/Core/Managed/Shared/Channel/TelemetryBuffer.cs

```diff
@@ -46,7 +48,30 @@ public int Capacity
                 this.capacity = value;
             }
         }
-        
+
+        /// <summary>
+        /// Gets or sets the maximum number of telemetry items that can be in the backlog to send. Items will be dropped
+        /// once this limit is hit.
+        /// </summary>        
+        public int MaximumUnsentBacklogSize
+        {
+            get
+            {
+                return this.maximumUnsentBacklogSize;
+            }
+
+            set
+            {
+                if (value < 1)
+                {
+                    this.maximumUnsentBacklogSize = DefaultMaximumUnsentBacklogSize;
```

Comment:
I would not swallow the user's value and set something else.  Either accept their value or throw an exception.
Since this is new code I will argue that we do not need to be afraid of throwing an exception for bad configuration since no existing customers reference this we can't break them.  But giving the customer and API we don't honor and silently do something different is a bad design.